### ISSUE TITLE
[Question] Why is shouldRun determined in background job? 

### ISSUE THREAD

**Issue by CarsonF** [jobbyphp/jobby#45]
It seems like it would be way faster to determine which jobs should be started in main process (maybe aside from locking), then only run those background jobs. Thoughts?

Also: This would also have a huge savings on not serializing closures that don't need to run.

**Comment by lamemind** [comment]
+1
my jobby keeps spawning new php processes every minute... not to be rude, but it's a clear waste of resources.

**Comment by garethellis36** [comment]
I agree that this is worth serious consideration. I have been testing this myself and it seems to work well:

```php
<?php

class AppJobby extends Jobby
{
    public function run()
    {
        foreach ($this->jobs as $job => $config) {
            $cron = CronExpression::factory($config['schedule']);
            if ($cron->isDue()) {
                parent::run();
            }
        }
    }
}
```
Obviously this only works if your schedule is set as a cron string but it could be adapted for other use cases.

**Comment by hellogerard** [comment]
PR?

**Comment by garethellis36** [comment]
I've started looking at this, but I run on Windows and I'm running in to issues with the test suite before changing anything. See #61

**Comment by hellogerard** [comment]
Closed in #62
